lib/rouge/lexers/velocity.rb

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Velocity < TemplateLexer
+      title 'Velocity'
+      desc 'Generic `Velocity <http://velocity.apache.org/>`_ template lexer.'
+      tag 'velocity'
+      filenames '*.vm', '*.velocity', '*.fhtml'
+      mimetypes 'text/html+velocity'
+
+      IDENTIFIER = IDENTIFIER = /[a-zA-Z_]\w*/
+
+      def self.analyze_text(text)
+        rv = 0.0
+        if text =~ /#\{?macro\}?\(.*?\).*?#\{?end\}?/
+          rv += 0.25
+        end
+
+        if text =~ /#\{?if\}?\(.+?\).*?#\{?end\}?/
+          rv += 0.15
+        end
+
+        if text =~ /#\{?foreach\}?\(.+?\).*?#\{?end\}?/
+          rv += 0.15
+        end
+
+        if text =~ /\$\{?[a-zA-Z_]\w*(\([^)]*\))?(\.\w+(\([^)]*\))?)*\}?/
+          rv += 0.01
+        end
+
+        rv
+      end
+
+      state :root do
+        rule /[^{#$]+/, Other
```

Comment:
Instead of highlighting this as `Other`, I'd recommend delegating it to `parent`, a lexer provided by the `TemplateLexer` class that by default is HTML. So:

``` ruby
rule /.../ do
  delegate parent
end
```